pkg/agent/agent.go

```diff
@@ -319,7 +319,12 @@ func getAgentResource() (*apiV1.ResourceInstance, error) {
 
 	response, err := agent.apicClient.ExecuteAPI(coreapi.GET, agentResourceURL, nil, nil)
 	if err != nil {
```

Comment:
this is throwing my off...  why not just return nil, err if err!= nil...
then you can take the if agentResource out of the if and make it stand alone

Reply:
so this is what Kevin and I worked on together.  AWS and Edge set their resource type to AWSDA/TA, EdgeDA/TA.  If only a generic resource was created an error here is hit that the resource was not found.

We wanted a way to say, lets also check if the agent name give is a generic type before throwing that error.  So we set the type to empty and call this method again, returning the generic resource type call.